core/new-gui/src/app/workspace/component/navigation/navigation.component.html

```diff
@@ -6,11 +6,22 @@
           <i id="texera-tour-icon" class="fa fa-question-circle-o" aria-hidden="true"></i>
         </span>
     </button>
-    <button mat-raised-button color="accent" class="texera-navigation-run-button" (click)="onClickRun()">
-      <mat-spinner class="texera-navigation-loading-spinner" *ngIf="showSpinner" [diameter]="30"></mat-spinner>
-      <div *ngIf="!showSpinner" class="texera-navigation-run-text">
+    <button *ngIf="!running" mat-raised-button color="accent" class="texera-navigation-run-button" (click)="onClickRun()">
+      <span class="texera-navigation-run-text">
+          <i id="texera-run-icon" class="fa fa-play" aria-hidden="true"></i>
+          Run
+      </span>
+    </button>
+
+    <button *ngIf="running" mat-raised-button color="accent" class="texera-navigation-run-button" (click)="onClickPauseResume()">
+      <div *ngIf="!isPaused" class="texera-navigation-run-text">
+          <i id="texera-run-icon" class="fa fa-spinner fa-spin" aria-hidden="true"></i>
+        Pause
+      </div>
+
+      <div *ngIf="isPaused" class="texera-navigation-run-text">
         <i id="texera-run-icon" class="fa fa-play" aria-hidden="true"></i>
```

Comment:
the icon ID should be different for all buttons, currently its all `texera-run-icon`. For a HTML document, an ID is an unique identifier for each of the HTML element, therefore, no duplicates should be allow in the html file.

Change pause to `texera-pause-icon` and resume to `texera-resume-icon`